File: DKPlugins/DKDom/DKDomUIEvent.js
// https://developer.mozilla.org/en-US/docs/Web/API/UIEvent

var UIEvent = function UIEvent(pointer) {
    // Properties
    /*
	Object.defineProperty(this, "cancelBubble", { //Not standardized, Deprecated
		get: function cancelBubble(){ 
			return CPP_DKDomUIEvent_cancelBubble(pointer)
		}
		set: function cancelBubble(flag){ 
			return CPP_DKDomUIEvent_cancelBubble(pointer, flag)
		}
	})
	*/
    Object.defineProperty(this, "detail", {
        //Read only
        get: function detail() {
            return CPP_DKDomUIEvent_detail(pointer)
        }
    })
    Object.defineProperty(this, "isChar", {
        //Read only
        get: function isChar() {
            return CPP_DKDomUIEvent_isChar(pointer)
        }
    })
    Object.defineProperty(this, "layerX", {
        //Read only
        get: function layerX() {
            return CPP_DKDomUIEvent_layerX(pointer)
        }
    })
    Object.defineProperty(this, "layerY", {
        //Read only
        get: function layerY() {
            return CPP_DKDomUIEvent_layerY(pointer)
        }
    })
    /*
	Object.defineProperty(this, "pageX", { //Read only
		get: function pageX(){ 
			return CPP_DKDomUIEvent_pageX(pointer)
		}
	})
	Object.defineProperty(this, "pageY", { //Read only
		get: function pageY(){ 
			return CPP_DKDomUIEvent_pageY(pointer)
		}
	})
	*/
    Object.defineProperty(this, "sourceCapabilities", {
        //Read only
        get: function sourceCapabilities() {
            return CPP_DKDomUIEvent_sourceCapabilities(pointer)
        }
    })
    Object.defineProperty(this, "view", {
        //Read only
        get: function view() {
            return CPP_DKDomUIEvent_view(pointer)
        }
    })
    /*
	Object.defineProperty(this, "which", { //Read only
		get: function which(){ 
			return CPP_DKDomUIEvent_which(pointer)
		}
	})
	*/

    // Methods
    UIEvent.prototype.initUIEvent = function initUIEvent() {
        //Deprecated
        CPP_DKDomUIEvent_initUIEvent(this.pointer)
    }

    return Event.call(this, pointer)
}
UIEvent.prototype = Event.prototype

File: DKPlugins/DKDom/DKDomUIEvent.cpp
// https://developer.mozilla.org/en-US/docs/Web/API/UIEvent

#include "DK/DKApp.h"
#include "DKDom/DKDomUIEvent.h"
#include "DKDom/DKDomEvent.h"
#include "DKRml/DKRml.h"


/////////////////////////
bool DKDomUIEvent::Init()
{
	DKDEBUGFUNC();

	// Properties
	//DKDuktape::AttachFunction("DKDomUIEvent_cancelBubble", DKDomUIEvent::cancelBubble);

	// Methods
	DKDuktape::AttachFunction("DKDomUIEvent_initUIEvent", DKDomUIEvent::initUIEvent);

	DKClass::DKCreate("DKDom/DKDomUIEvent.js");
	return true;
}


// Properties
/*
///////////////////////////////////////////////
int DKDomUIEvent::cancelBubble(duk_context* ctx)
{
	DKDEBUGFUNC(ctx);
	DKString eventAddress = duk_require_string(ctx, 0);
	Rml::Event* event = DKEvent::addressToEvent(eventAddress);
	if (!event) {
		DKERROR("DKDomUIEvent::cencelBubble(): event invalid\n");
		duk_push_boolean(ctx, false);
		return true;
	}
	bool flag = duk_require_boolean(ctx, 1);
	if(flag){
		event->StopPropagation();
		return true;
	}
	return true;
}
*/

// Methods
///////////////////////////////////////////////
int DKDomUIEvent::initUIEvent(duk_context* ctx)
{
	DKDEBUGFUNC(ctx);
	DKString eventAddress = duk_require_string(ctx, 0);
	Rml::Event* event = DKRml::addressToEvent(eventAddress);
	if (!event) {
		DKERROR("DKDomUIEvent::initEvent(): event invalid\n");
		duk_push_boolean(ctx, false);
		return true;
	}

	//TODO
	return false;
}